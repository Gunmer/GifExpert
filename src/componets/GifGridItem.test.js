import {GifGridItem} from "./GifGridItem";
import {shallow} from "enzyme";

describe('GifGridItem', () => {
  const title = 'My Title';
  const url = 'http://my-title.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  it('should be render', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should have p with title', () => {
    const p = wrapper.find('p');

    expect(p.text().trim()).toBe(title);
  });

  it('should have img and alt', () => {
    const img = wrapper.find('img');

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  it('should have animate__flipInX', () => {
    const div = wrapper.find('div');

    expect(div.prop('className')).toContain('animate__flipInX');
  });
});
