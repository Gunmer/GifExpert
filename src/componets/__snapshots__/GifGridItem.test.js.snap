// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem should be render 1`] = `
<div
  className="card animate__animated animate__flipInX"
>
  <img
    alt="My Title"
    src="http://my-title.jpg"
  />
  <p>
    My Title
  </p>
</div>
`;
